ThreadPool.h

```diff
@@ -12,11 +12,15 @@
 #include <vector>
 
 class ThreadPool {
+  public:
+  static void* entry_point(void* context);
+
  public:
-  explicit ThreadPool(std::size_t threads);
+  explicit ThreadPool(std::size_t num_threads);
 
   template <class F>
-  ThreadPool(std::size_t threads, F&& initialize);
+  ThreadPool(std::size_t num_threads, F &&initialize,
+             std::size_t stack_size = 0);
```

Comment:
the `stack_size` of `0` means use the default platform size. this should allow this class to continue to operate in the same way it did previously, but now we can use this class with an explicit sized manner.